samcli/lib/delete/cfn_utils.py

```diff
@@ -116,8 +116,10 @@ def wait_for_delete(self, stack_name):
 
         # Wait for Delete to Finish
         waiter = self._client.get_waiter("stack_delete_complete")
-        # Poll every 30 seconds and set max attempts to be 3.
-        waiter_config = {"Delay": 30, "MaxAttempts": 3}
+        # Poll every 30 seconds which is the default, but just explicitly called out.
+        # Do not set `MaxAttempts` in waiter_config - regression: https://github.com/aws/aws-sam-cli/issues/4361
+        # Waiter configuration docs at: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudformation.html#waiters
+        waiter_config = {"Delay": 30}
```

Comment:
If we are using the default, why do we need to put in a definition here? If you were being explicit about `Delay`, should we also be explicit about the MaxAttempt (`120` by default) ?

Reply:
It was a point of education for ourselves, however I have instead removed those lines and just added a link to the regression issue.

Reply:
Looking at the commit history, it was initially set to 5 seconds and changed to 30 later in this [PR](https://github.com/aws/aws-sam-cli/pull/3232) .